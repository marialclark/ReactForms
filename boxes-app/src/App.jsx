import BoxList from './BoxList';
import './styles/App.css';

function App() {
  return (
    <div className="App">
      <h1>Color Box Maker</h1>
      <BoxList />
    </div>
  );
}

export default App;
