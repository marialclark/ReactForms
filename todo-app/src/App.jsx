import TodoList from './TodoList';
import './styles/app.css';

function App() {
  return (
    <div className="App">
      <h1>Todo App</h1>
      <TodoList />
    </div>
  );
}

export default App;