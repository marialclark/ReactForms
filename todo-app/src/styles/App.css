/* Center the app and style the header */
.App {
  text-align: center;
  font-family: Arial, sans-serif;
  margin: 20px;
}

h1 {
  color: #333;
  font-size: 24px;
}
