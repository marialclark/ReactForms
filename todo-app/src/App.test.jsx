import { render, screen } from '@testing-library/react';
import App from './App';

describe('App Component', () => {
  // Smoke Test
  it('renders without crashing', () => {
    render(<App />);
  });

  // Snapshot Test
  it('matches snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  // Critical Business Logic: Content Render Test
  it('renders the Todo App heading', () => {
    render(<App />);
    expect(screen.getByText('Todo App')).toBeInTheDocument();
  });
});
