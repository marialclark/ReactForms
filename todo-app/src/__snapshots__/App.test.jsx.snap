// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      Todo App
    </h1>
    <div
      class="TodoList"
    >
      <div
        class="NewTodoForm"
      >
        <form>
          <label
            for="task"
          >
            Task:
          </label>
          <input
            id="task"
            name="task"
            type="text"
            value=""
          />
          <button>
            Add Todo
          </button>
        </form>
      </div>
      <div
        class="TodoList-container"
      />
    </div>
  </div>
</DocumentFragment>
`;
